pyworker/worker.py

```diff
@@ -46,6 +60,42 @@ def signal_handler(signum, frame):
         signal.signal(signal.SIGTERM, signal_handler)
         yield
 
+    @contextmanager
+    def _instrument(self, job, start_time):
+
+        def _latency(run_at, ran_at):
+
+            # Specify UTC timezone and convert to unix time
+            run_at_utc = run_at.replace(tzinfo=timezone.utc)
+            run_at_time = run_at_utc.timestamp()
+
+            # Difference between when the job was scheduled `run_at`
+            # and when the job actually started running `ran_at`
+            return ran_at - run_at_time
+
+        if self.newrelic_app:
+
+            latency = _latency(job.run_at, start_time)
+
+            with newrelic.agent.BackgroundTask(
+                    application=self.newrelic_app,
+                    name='%s#run' % job.class_name,
+                    group='DelayedJob') as task:
+
+                # Record a custom metrics
+                # 1) Custom/DelayedJobQueueLatency/<job.queue> => latency
+                # 2) Custom/DelayedJobLatency/<job.name> => latency
+                # 3) Custom/DelayedJobAttempts/<job.name> => attempts
+                newrelic.agent.record_custom_metrics([
+                    ('Custom/DelayedJobQueueLatency/%s' % job.queue, latency),
+                    ('Custom/DelayedJobLatency/%s' % job.class_name, latency),
+                    ('Custom/DelayedJobAttempts/%s' % job.class_name, job.attempts + 1)  # Add 1 since it 0-index
```

Comment:
Our convention is to report previous attempts so no need to add 1 here.

Reply:
Alright, removed